wolfcrypt/src/port/psa/psa_pkcbs.c

```diff
@@ -0,0 +1,556 @@
+/* psa_pkcbs.c
+ *
+ * Copyright (C) 2006-2021 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#include <wolfssl/wolfcrypt/settings.h>
+
+#if defined(WOLFSSL_HAVE_PSA)
+#if defined(HAVE_PK_CALLBACKS)
+#if !defined(WOLFSSL_PSA_NO_PKCBS)
+
+#include <psa/crypto.h>
+
+#include <wolfssl/wolfcrypt/port/psa/psa.h>
+#include <wolfssl/wolfcrypt/error-crypt.h>
+#include <wolfssl/internal.h>
+#include <wolfssl/wolfcrypt/asn.h>
+#include <wolfssl/wolfcrypt/logging.h>
+
+#ifdef HAVE_ECC
+
+#include <wolfssl/wolfcrypt/ecc.h>
+
+static int psa_ecc_get_curve_info_from_curve_id(int ecc_curve,
+                                                psa_ecc_family_t *family,
+                                                size_t *key_size)
+{
+    if (ecc_curve != ECC_SECP256R1)
+        return WOLFSSL_NOT_IMPLEMENTED;
+
+    *family = PSA_ECC_FAMILY_SECP_R1;
+    if (key_size != NULL)
+        *key_size = 32;
+
+    return 0;
+}
+
+/**
+ * psa_ecc_export_to_wc_key() - export psa @key_id into ecc_key @key
+ * @key: already initialized ecc_key to export the key into [out]
+ * @key_id: key to export [in]
+ * @key_size: size fo the key [in]
+ * @curve_id: wc id of the curve [in]
+ *
+ * returns 0 on success
+ */
+static int psa_ecc_export_to_wc_key(struct ecc_key *key,
+                                    psa_key_id_t key_id,
+                                    size_t key_size,
+                                    int curve_id)
+{
+    uint8_t raw[ECC_MAXSIZE * 2];
+    psa_status_t status;
+    size_t length;
+    int ret;
+
+    PSA_LOCK();
+    status = psa_export_public_key(key_id,
+                                   raw, ECC_MAXSIZE * 2, &length);
+    PSA_UNLOCK();
+
+    if (status != PSA_SUCCESS) {
+        WOLFSSL_MSG("PSA: export key error");
+        return WC_HW_E;
+    }
+
+    ret = wc_ecc_import_unsigned(key, raw + 1,
```

Comment:
Please use `int wc_ecc_import_x963_ex(const byte* in, word32 inLen, ecc_key* key, int curve_id)`. This will make sure all point types (compress odd/even) are all supported correctly.